### ISSUE TITLE
转换自己的weights成功后，执行example时候报错ncnn error:-100

### ISSUE THREAD

**Issue by Aionrichman** [xiangweizeng/darknet2ncnn#23]
使用工程自带的weights是可以成功转换并跑example的。使用自己训练的单分类yolo也能转换成功，但是跑example时报错ncnn error:-100

**Comment by xiangweizeng** [comment]
返回-100， 这里应该是没有检测到目标

**Comment by Aionrichman** [comment]
看了一下，是我这边的类别名没有设置对。设置好后跑example，报错Segmentation fault (core dumped)了

**Comment by xiangweizeng** [comment]
具体问题描述， 欢迎进入QQ群  852287556 交流

**Comment by Aionrichman** [comment]
嗯嗯，加了。这边贴一下我的yolo cfg和param

yolo--------------------------------------------------------------------------------------------------
[net]
# Testing
batch=1
subdivisions=1
# Training
# batch=64
# subdivisions=2
width=416
height=416
channels=3
momentum=0.9
decay=0.0005
angle=0
saturation = 1.5
exposure = 1.5
hue=.1

learning_rate=0.001
burn_in=1000
max_batches = 500200
policy=steps
steps=400000,450000
scales=.1,.1

[convolutional]
batch_normalize=1
filters=16
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=32
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=64
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=128
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=256
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=512
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=1

[convolutional]
batch_normalize=1
filters=1024
size=3
stride=1
pad=1
activation=leaky

###########

[convolutional]
batch_normalize=1
filters=256
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=512
size=3
stride=1
pad=1
activation=leaky

[convolutional]
size=1
stride=1
pad=1
filters=18
activation=linear



[yolo]
mask = 3,4,5
anchors = 10,14,  23,27,  37,58,  81,82,  135,169,  344,319
classes=1
num=6
jitter=.3
ignore_thresh = .7
truth_thresh = 1
random=1

[route]
layers = -4

[convolutional]
batch_normalize=1
filters=128
size=1
stride=1
pad=1
activation=leaky

[upsample]
stride=2

[route]
layers = -1, 8

[convolutional]
batch_normalize=1
filters=256
size=3
stride=1
pad=1
activation=leaky

[convolutional]
size=1
stride=1
pad=1
filters=18
activation=linear

[yolo]
mask = 0,1,2
anchors = 10,14,  23,27,  37,58,  81,82,  135,169,  344,319
classes=1
num=6
jitter=.3
ignore_thresh = .7
truth_thresh = 1
random=1

--------------------------------------------------------------------------------------
7767517
50 52
Input            data 0 1 data 0=416 1=416 2=3
Convolution      conv_0 1 1 data conv_0 0=16 1=3 2=1 3=1 4=1 5=0 6=432
BatchNorm        conv_0_batch_norm 1 1   conv_0   conv_0_batch_norm 0=16 1=0.00001
ReLU             conv_0_activation 1 1 conv_0_batch_norm conv_0_activation 0=0.1
Pooling          maxpool_1 1 1 conv_0_activation maxpool_1 0=0 1=2 2=2 3=0 5=1 13=0 14=1 15=1
Convolution      conv_2 1 1 maxpool_1 conv_2 0=32 1=3 2=1 3=1 4=1 5=0 6=4608
BatchNorm        conv_2_batch_norm 1 1   conv_2   conv_2_batch_norm 0=32 1=0.00001
ReLU             conv_2_activation 1 1 conv_2_batch_norm conv_2_activation 0=0.1
Pooling          maxpool_3 1 1 conv_2_activation maxpool_3 0=0 1=2 2=2 3=0 5=1 13=0 14=1 15=1
Convolution      conv_4 1 1 maxpool_3 conv_4 0=64 1=3 2=1 3=1 4=1 5=0 6=18432
BatchNorm        conv_4_batch_norm 1 1   conv_4   conv_4_batch_norm 0=64 1=0.00001
ReLU             conv_4_activation 1 1 conv_4_batch_norm conv_4_activation 0=0.1
Pooling          maxpool_5 1 1 conv_4_activation maxpool_5 0=0 1=2 2=2 3=0 5=1 13=0 14=1 15=1
Convolution      conv_6 1 1 maxpool_5 conv_6 0=128 1=3 2=1 3=1 4=1 5=0 6=73728
BatchNorm        conv_6_batch_norm 1 1   conv_6   conv_6_batch_norm 0=128 1=0.00001
ReLU             conv_6_activation 1 1 conv_6_batch_norm conv_6_activation 0=0.1
Pooling          maxpool_7 1 1 conv_6_activation maxpool_7 0=0 1=2 2=2 3=0 5=1 13=0 14=1 15=1
Convolution      conv_8 1 1 maxpool_7 conv_8 0=256 1=3 2=1 3=1 4=1 5=0 6=294912
BatchNorm        conv_8_batch_norm 1 1   conv_8   conv_8_batch_norm 0=256 1=0.00001
ReLU             conv_8_activation 1 1 conv_8_batch_norm conv_8_activation 0=0.1
Split            conv_8_activation_split 1 2 conv_8_activation  conv_8_activation_split_0 conv_8_activation_split_1
Pooling          maxpool_9 1 1 conv_8_activation_split_0 maxpool_9 0=0 1=2 2=2 3=0 5=1 13=0 14=1 15=1
Convolution      conv_10 1 1 maxpool_9 conv_10 0=512 1=3 2=1 3=1 4=1 5=0 6=1179648
BatchNorm        conv_10_batch_norm 1 1   conv_10   conv_10_batch_norm 0=512 1=0.00001
ReLU             conv_10_activation 1 1 conv_10_batch_norm conv_10_activation 0=0.1
Pooling          maxpool_11 1 1 conv_10_activation maxpool_11 0=0 1=2 2=1 3=0 5=1 13=0 14=1 15=1
Convolution      conv_12 1 1 maxpool_11 conv_12 0=1024 1=3 2=1 3=1 4=1 5=0 6=4718592
BatchNorm        conv_12_batch_norm 1 1   conv_12   conv_12_batch_norm 0=1024 1=0.00001
ReLU             conv_12_activation 1 1 conv_12_batch_norm conv_12_activation 0=0.1
Convolution      conv_13 1 1 conv_12_activation conv_13 0=256 1=1 2=1 3=1 4=0 5=0 6=262144
BatchNorm        conv_13_batch_norm 1 1   conv_13   conv_13_batch_norm 0=256 1=0.00001
ReLU             conv_13_activation 1 1 conv_13_batch_norm conv_13_activation 0=0.1
Split            conv_13_activation_split 1 2 conv_13_activation  conv_13_activation_split_0 conv_13_activation_split_1
Convolution      conv_14 1 1 conv_13_activation_split_0 conv_14 0=512 1=3 2=1 3=1 4=1 5=0 6=1179648
BatchNorm        conv_14_batch_norm 1 1   conv_14   conv_14_batch_norm 0=512 1=0.00001
ReLU             conv_14_activation 1 1 conv_14_batch_norm conv_14_activation 0=0.1
Convolution      conv_15 1 1 conv_14_activation conv_15 0=18 1=1 2=1 3=1 4=0 5=1 6=9216
DarknetActivation conv_15_activation 1 1 conv_15 conv_15_activation 0=3
Concat           route_17 1 1  conv_13_activation_split_1 route_17 0=0
Convolution      conv_18 1 1 route_17 conv_18 0=128 1=1 2=1 3=1 4=0 5=0 6=32768
BatchNorm        conv_18_batch_norm 1 1   conv_18   conv_18_batch_norm 0=128 1=0.00001
ReLU             conv_18_activation 1 1 conv_18_batch_norm conv_18_activation 0=0.1
Interp           upsample_19 1 1 conv_18_activation upsample_19 0=1 1=2.f 2=2.f
Concat           route_20 2 1  upsample_19 conv_8_activation_split_1 route_20 0=0
Convolution      conv_21 1 1 route_20 conv_21 0=256 1=3 2=1 3=1 4=1 5=0 6=884736
BatchNorm        conv_21_batch_norm 1 1   conv_21   conv_21_batch_norm 0=256 1=0.00001
ReLU             conv_21_activation 1 1 conv_21_batch_norm conv_21_activation 0=0.1
Convolution      conv_22 1 1 conv_21_activation conv_22 0=18 1=1 2=1 3=1 4=0 5=1 6=4608
DarknetActivation conv_22_activation 1 1 conv_22 conv_22_activation 0=3
Yolov3Detection  yolo_23 2 1  conv_15_activation conv_22_activation yolo_23 0=1 1=6 2=0 3=0.25f 4=0.45f 5=416 6=416 -23307=12,81.000000,82.000000,135.000000,169.000000,344.000000,319.000000,10.000000,14.000000,23.000000,27.000000,37.000000,58.000000

**Comment by Aionrichman** [comment]
非常感谢大神的解答，Segmentation fault (core dumped)的原因是label里面没有设置background，增加后能正常跑example了

**Comment by wanzghen0123** [comment]
> bac

你好，请问是什么意思啊？我转换自己的模型后，报错：段错误 (核心已转储)

**Comment by Aionrichman** [comment]
转换成功了，但是做inference的时候报错

**Comment by kjcx100** [comment]
> 非常感谢大神的解答，Segmentation fault (core dumped)的原因是label里面没有设置background，增加后能正常跑example了

搞不懂为啥直接段错误了，该提示下label的问题。。。虽然加了background后正常了，但是直接用darknet不用加没这个问题。

**Comment by xiangweizeng** [comment]
你好，darknet的实现里默认没有背景，ncnn实现有背景，将类别ID加一了，保留给背景使用，所以需要加入背景类。


 
---原始邮件---
发件人: "kjcx100"<notifications@github.com&gt;
发送时间: 2020年4月11日 15:59:31
收件人: "xiangweizeng/darknet2ncnn"<darknet2ncnn@noreply.github.com&gt;;
抄送: "xiangweizeng"<826127212@qq.com&gt;;"Comment"<comment@noreply.github.com&gt;;
主题: Re: [xiangweizeng/darknet2ncnn] 转换自己的weights成功后，执行example时候报错ncnn error:-100 (#23)




  
非常感谢大神的解答，Segmentation fault (core dumped)的原因是label里面没有设置background，增加后能正常跑example了
  
搞不懂为啥直接段错误了，该提示下label的问题。。。虽然加了background后正常了，但是直接用darknet不用加没这个问题。
